spec/predicates/runtime-trace.md

```diff
@@ -0,0 +1,2677 @@
+
+This attestation can be used in conjunction with provenance attestations to leverage observability to attest to various SLSA requirements. The runtime trace can prove the build was invoked via a script, that the build was executed in a hermetic environment with no network access, and so on.
+
+## Schema
+
+The schema of the attestation type is the core part of the document. It defines the fields included in an instantiation of the attestation type, as well as rules to parse them.
+
+```json
+{
+  "_type": "https://in-toto.io/Statement/v0.1",
+  "predicateType": "https://in-toto.io/runtime-trace/v0.1.0",
+  "subject": [{ ... }],
+  "predicate": {
+    "monitor": {
+      "id": "<URI>"
+    },
+    “build”: {
+        “builderId”: “<URI>”,
+        “type”: “<URI>”,
+        “event”: “<STRING>”,
+    },
+    "invocation": {
+     "configSource": {
+        "uri": "<URI>",
+        "digest": { /* DigestSet */ },
+        "entryPoint": "<STRING>"
+      },
+      "tracePolicy": {}
+    },
+    "monitorLog": {
+      "process": [
+         { /* object */ }
+      ],
+      "fileAccess": [
+        { /* object */ }
+      ],
+    },
+    "metadata": {
+      "buildStartedOn": "<TIMESTAMP>",
+      "buildFinishedOn": "<TIMESTAMP>",
+    }
+  }
+}
+```
+
+### Parsing Rules
+
+This predicate follows the in-toto attestation parsing rules. Summary:
+
+-   Consumers MUST ignore unrecognized fields.
+-   The `predicateType` URI includes the major version number and will always change whenever there is a backwards incompatible change.
+-   Minor version changes are always backwards compatible and “monotonic.” Such changes do not update the `predicateType`.
+-   Producers MAY add extension fields using field names that are URIs.
+-   _optional_ fields MAY be unset or null, and should be treated equivalently. Both are equivalent to empty for object or array values.
+
+## Fields
+
+`monitor` _object_, _required_
+
+Identifies the specific monitor instance used to trace the runtime.
+
+`monitor.id` _string (URI)_, _required_
+
+URI indicating the monitor’s identity.
+
+`build` _object_, _required_
+
+Feedback required: Should this predicate be scoped to runtime traces of builds only or generalize to runtime traces of any activity?
+
+`builder.builderId` _string (URI)_, _required_
+
+URI indicating the builder’s identity. Ex: GitLab Runner ID
+
+`build.type` _string (TypeURI)_, _required_
+
+URI indicating the type of build performed.
+
+`build.event` _string_, _required_
+
+String identifying the specific job or task associated with the attestation.
+
+`invocation` _object_, _optional_
+
+Identifies the event that initiated the monitor. Similar to `invocation` in SLSA Provenance.
+
+`invocation.configSource` _object_, _optional_
+
+Effectively a pointer to the monitor's configuration.
+
+`invocation.tracePolicy` _object_, _optional_
```

Comment:
IMO `tracePolicy` should move under `monitor`. Different tracers have different features that might not be available across the board. also the syntax will definitely be different between tracers and it will make most sense to keep the trace policy in it's original form. so if the tracePolicy is monitor-specific it's clearer to move it there.

Reply:
+1

Reply:
@pxp928

Reply:
I think we can move all of invocation's fields into the monitor object. WDYT @pxp928?